amp.php

```diff
@@ -145,6 +154,31 @@ function amp_maybe_add_actions() {
 	}
 }
 
+/**
+ * Whether this is in 'canonical mode.'
+ *
+ * Themes can register support for this with `add_theme_support( 'amp' )`.
+ * Then, this will change the plugin from 'paired mode,' and it won't use its own templates.
+ * Nor output frontend markup like the 'rel' link. If the theme registers support for AMP with:
+ * `add_theme_support( 'amp', array( 'template_path' => get_template_directory() . 'my-amp-templates/' ) )`
+ * it will retain 'paired mode.
+ *
+ * @return boolean Whether this is in AMP 'canonical mode'.
+ */
+function amp_is_canonical() {
+	$support = get_theme_support( 'amp' );
+	if ( true === $support ) {
+		return true;
+	}
+	if ( is_array( $support ) ) {
+		$args = array_shift( $support );
+		if ( empty( $args['template_path'] ) ) {
```

Comment:
The use of `template_path` will probably need to be iterated upon. For example, `theme_root` per `\WP_Theme::get_theme_root()` or `template_dir` per `\WP_Theme::get_template_directory()`. This will need some more discovery in #849 as to what makes the most sense. To me it seems the `theme_root` and `template_directory` are the same. /cc @kaitnyl